### ISSUE TITLE
cargo install error cannot infer type of the type parameter `P` declared on the associated function `strip_prefix`

### ISSUE THREAD

**Issue by usrtax** [watchexec/watchexec#506]
❯ cargo --version
cargo 1.67.1 (8ecd4f20a 2023-01-10)
~/docker_dev_build main*                                                                                                         root@con-nrnberg-62-171-170-189 13:34:03
❯ cargo install watchexec-cli

<img width="1728" alt="image" src="https://user-images.githubusercontent.com/110233389/220828313-bc1ac94e-fd65-4d61-9003-012fda5873d4.png">

```
    = note: multiple `impl`s satisfying `BStr: AsRef<_>` found in the `bstr` crate:
            - impl AsRef<BStr> for BStr;
            - impl AsRef<[u8]> for BStr;
help: consider specifying the generic argument
    |
139 |             name.strip_prefix::<P>(Category::LinkedPseudoRef { name: "".into() }.prefix().as_ref())
    |                              +++++

error[E0283]: type annotations needed
  --> /opt/rust/registry/src/github.com-1ecc6299db9ec823/git-ref-0.22.0/src/raw.rs:76:18
   |
76 |                 .strip_prefix(namespace.0.as_bstr().as_ref())
   |                  ^^^^^^^^^^^^ ------------------------------ type must be known at this point
   |                  |
   |                  cannot infer type of the type parameter `P` declared on the associated function `strip_prefix`
   |
   = note: cannot satisfy `_: core::slice::SlicePattern`
note: required by a bound in `core::slice::<impl [T]>::strip_prefix`
  --> /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/core/src/slice/mod.rs:2284:28
   |
   = note: required by this bound in `core::slice::<impl [T]>::strip_prefix`
help: consider specifying the generic argument
   |
76 |                 .strip_prefix::<P>(namespace.0.as_bstr().as_ref())
   |                              +++++

error[E0283]: type annotations needed
  --> /opt/rust/registry/src/github.com-1ecc6299db9ec823/git-ref-0.22.0/src/raw.rs:76:18
   |
76 |                 .strip_prefix(namespace.0.as_bstr().as_ref())
   |                  ^^^^^^^^^^^^                       ------ type must be known at this point
   |                  |
   |                  cannot infer type of the type parameter `P` declared on the associated function `strip_prefix`
   |
   = note: multiple `impl`s satisfying `BStr: AsRef<_>` found in the `bstr` crate:
           - impl AsRef<BStr> for BStr;
           - impl AsRef<[u8]> for BStr;
help: consider specifying the generic argument
   |
76 |                 .strip_prefix::<P>(namespace.0.as_bstr().as_ref())
   |                              +++++

Some errors have detailed explanations: E0277, E0283.
For more information about an error, try `rustc --explain E0277`.
error: could not compile `git-ref` due to 8 previous errors
warning: build failed, waiting for other jobs to finish...
error: failed to compile `watchexec-cli v1.21.1`, intermediate artifacts can be found at `/tmp/cargo-install5bAQsj`
```

**Comment by kesyog** [comment]
Try `cargo install --locked watchexec-cli` as a workaround

**Comment by passcod** [comment]
Thanks @kesyog, that's indeed what you should do. I'll look into locking down the dependencies to avoid this error for the current while.

**Comment by passcod** [comment]
Just to mention I'm working on releasing a new version to solve this but have been very busy at dayjob.

**Comment by passcod** [comment]
This should be solved with 1.22.0.
